### PR TITLE
Fix 1285

### PR DESCRIPTION
**Pull request by kminehart**
## Related issue

#1285 

## Proposed changes

Summary: Changed the defaults for `Name`, `Description`, and `Hint` from `consent/types.go`.

Changes:

1. Changed the `ToRFCError()` function to only provide defaults if all 3 (Name, Description, and Hint) are left blank.
    * I think it's safe to assume that if a developer provides any combination of "Name", "Description", or "Hint", injecting any defaults will cause confusion.
2. Changed the default values of `Name`, `Description`, and `Hint` to be more generic, and clear that they were left blank.
    * Name: "Consent request denied"
    * Description: "The request was denied and no further description was provided"
    * Given no context at the time of generation, no valuable hint could be provided, so it will remain empty.

## Checklist

- [x] I have read the [contributing guidelines](CONTRIBUTING.md)
- [x] I confirm that this pull request does not address a security vulnerability. If this pull request addresses a security
vulnerability, I confirm that I got green light (please contact [hi@ory.sh](mailto:hi@ory.sh)) from the maintainers to push the changes.
- [x] I signed the [Developer's Certificate of Origin](https://github.com/ory/keto/blob/master/CONTRIBUTING.md#developers-certificate-of-origin)
by signing my commit(s). You can amend your signature to the most recent commit by using `git commit --amend -s`. If you
amend the commit, you might need to force push using `git push --force HEAD:<branch>`. Please be very careful when using
force push.
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation within the code base (if appropriate)
- [x] I have documented my changes in the [developer guide](https://github.com/ory/docs) (if appropriate)

### CODE REVIEW

**Comment by aeneasr** [comment]
I don't get why crypto is failing all the time. You can fix this with `go mod tidy` though!

File: consent/types.go [hunk 1]
```diff
@@ -27,6 +27,11 @@ import (
 	"github.com/ory/hydra/client"
 )
 
+const (
+	requestDeniedErrorName        = "Consent request denied"
```
Comment (aeneasr) [inline consent/types.go]: Errors names [should not start with a capital letter](https://github.com/golang/go/wiki/Errors) :)

File: consent/types.go [hunk 2]
```diff
@@ -53,18 +58,14 @@ type RequestDeniedError struct {
...
 		e.Code = fosite.ErrInvalidRequest.Code
 	}
-	if e.Description == "" {
-		e.Description = fosite.ErrInvalidRequest.Description
-	}
-	if e.Hint == "" {
-		e.Hint = fosite.ErrInvalidRequest.Hint
-	}
 	if e.Debug == "" {
 		e.Debug = fosite.ErrInvalidRequest.Debug
```
Comment (aeneasr) [inline consent/types.go]: Probably makes sense to remove this too?

File: consent/types.go (hunk 1 above)
Comment (kminehart) [reply consent/types.go]: I like capitalization too much.

Fixed.

File: consent/types.go (hunk 2 above)
Comment (kminehart) [reply consent/types.go]: Yeah I think so.

Should the default error code stay as `fosite.ErrInvalidRequest` (400)?

**Comment by aeneasr** [comment]
Alternatively `go get -u  golang.org/x/crypto` with go modules enabled.

**Comment by kminehart** [comment]
My bad, go modules are a new thing for me. I ran `go mod tidy` and committed the `go.mod` and `go.sum` files. That should suffice, right?

**Comment by aeneasr** [comment]
Perfect, last thing would be to add a test so this doesn't break in a future patch by accident :)

**Comment by kminehart** [comment]
Cool. I wasn't sure what the right place was to put the unit tests for this change, or if they were necessary given the nature of the tests in `hydra/consent`.

I also am not sure if there needs to be any documentation changes. I couldn't find anything relevant but I could have missed something.

File: consent/types.go [hunk 3]
```diff
@@ -27,6 +27,11 @@ import (
 	"github.com/ory/hydra/client"
 )
 
+const (
+	requestDeniedErrorName        = "consent request denied"
+	requestDeniedErrorDescription = "the request was denied and no further description was provided"
```
Comment (aeneasr) [inline consent/types.go]: The description can be upper case :)

File: consent/types_test.go [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+	for k, tc := range []struct {
+		input  *RequestDeniedError
+		expect *fosite.RFC6749Error
+	}{
+		{
+			input: &RequestDeniedError{
+				Description: "not empty",
+			},
+			expect: &fosite.RFC6749Error{
+				Name:        "",
```
Comment (aeneasr) [inline consent/types_test.go]: Wouldn't it make sense for the name and description to always be set to the default values? Then you can override e.g. just the description or only the name without having to set both?

File: consent/types.go (hunk 3 above)
Comment (kminehart) [reply consent/types.go]: ahhhh!

**Comment by kminehart** [comment]
@aeneasr I think this one is ready.

Pretty basic, if `Name` is not provided, it'll use the default. `Description` and `Hint` no longer have defaults.  All of the commits were me just getting used to the way everything is structured. 😅

**Comment by aeneasr** [comment]
Thank you!

**Pull request merged**